packages/shared/src/types/analysis.ts

```diff
@@ -45,3 +48,26 @@ export interface AnalysisInput {
  * A common interface for all kinds of analysis output.
  */
 export interface AnalysisOutput {}
+
+/**
+ * In SonarQube context, an analysis input includes both path and content of a file
+ * to analyze. However, in SonarLint, we might only get the file path. As a result,
+ * we read the file if the content is missing in the input.
+ */
+export async function fillContent<T extends AnalysisInput>(
```

Comment:
mentioning the exact field you want to fill is a bit more explicit.
```suggestion
export async function fillFileContent<T extends AnalysisInput>(
```